### PR TITLE
Add request header param in the access and dump log

### PR DESCRIPTION
**Pull request by tanvir-retailai**

### CODE REVIEW

**Review by kongliang1 — APPROVED** [review]
`v := c.Request().Header.Get(param)
							if v != "" {
								reqHeader[param] = c.Request().Header.Get(param)
							}`

Duplicate value,can we change it like:

`v := c.Request().Header.Get(param)
							if v != "" {
								reqHeader[param] = v
							}`

File: middleware/access_log.go [hunk 1]
```diff
@@ -247,12 +250,34 @@ func AccessLoggerWithConfig(config LoggerConfig) echo.MiddlewareFunc {
...
+						return buf.WriteString(res_body)
+					}
+					return buf.WriteString(`null`)
+				case "req_header":
+					if len(config.RequestHeader) > 0 {
+						reqHeader := make(map[string]interface{})
+						for _, param := range config.RequestHeader {
+							v := c.Request().Header.Get(param)
+							if v != "" {
+								reqHeader[param] = c.Request().Header.Get(param)
```
Comment (zhongdalu-trechina) [inline middleware/access_log.go]: Maybe this 'reqHeader[param] = v' would be better.

File: middleware/access_log.go [hunk 2]
```diff
@@ -330,6 +355,21 @@ func (config LoggerConfig) logAccess(c echo.Context) (err error) {
 				cl = "0"
 			}
 			return buf.WriteString(cl)
+		case "req_header":
+			if len(config.RequestHeader) > 0 {
+				reqHeader := make(map[string]interface{})
+				for _, param := range config.RequestHeader {
+					v := c.Request().Header.Get(param)
+					if v != "" {
+						reqHeader[param] = c.Request().Header.Get(param)
```
Comment (zhongdalu-trechina) [inline middleware/access_log.go]: Maybe this 'reqHeader[param] = v' would be better.

File: middleware/access_log.go [hunk 3]
```diff
@@ -247,12 +250,34 @@ func AccessLoggerWithConfig(config LoggerConfig) echo.MiddlewareFunc {
 					return buf.WriteString(strconv.FormatInt(res.Size, 10))
 				case "request_body":
 					if len(reqBody) > 0 {
-						reqBody, _ = maskSensitiveInfo(reqBody, config.MaskedParameters)
-						return buf.Write(reqBody)
+						reqBody, err = maskSensitiveInfo(reqBody, config.MaskedParameters)
+						if err == nil {
+							return buf.Write(reqBody)
+						}
+						return buf.WriteString(`null`)
```
Comment (zhongdalu-trechina) [inline middleware/access_log.go:257]: I think delete it will be okay

**Comment by SunJinping-TRE-C** [comment]
Why not output all the request header parameters instead of set up `RequestHeader` in config? Sometime latter we maybe need the other parameters, who knows

**Comment by zhongdalu-trechina** [comment]
> Why not output all the request header parameters instead of set up `RequestHeader` in config? Sometime latter we maybe need the other parameters, who knows

then we have to mask some secret stuff. It's also a good way I think

**Comment by SunJinping-TRE-C** [comment]
> > Why not output all the request header parameters instead of set up `RequestHeader` in config? Sometime latter we maybe need the other parameters, who knows
> 
> then we have to mask some secret stuff. It's also a good way I think

Yes, just like `bodyDumpMaskParam`

**Review by zhongdalu-trechina — APPROVED** [review]

**Pull request merged**